### ISSUE TITLE
code for super == not generated correctly

### ISSUE THREAD

**Issue by jmesserly** [dart-lang/dev_compiler#226]
this affects package:source_span/src/file.dart

```dart
  bool operator ==(other) {
    if (other is! FileSpan) return super == other;
    return _start == other._start && _end == other._end &&
        sourceUrl == other.sourceUrl;
  }
```

The intent is to call the `super['=='](other)` but we generate this as `dart.equals(super, other)` which is not legal.

**Comment by jmesserly** [comment]
this is caught somehow transitively from package:test

**Comment by jmesserly** [comment]
https://codereview.chromium.org/1183453005
